### ISSUE TITLE
Custom Coderpad

### ISSUE THREAD

**Issue by tko22** [hack4impact-uiuc/h4i-recruitment#189]
¯\_(ツ)_/¯ We could make it run specific languages (maybe only get it to work with python first?) and then we could also add in intellisense as well... who knows. 

However, I would build out something that would allow a user to drag and drop a file with code (obviously with specific entry points), which would run in the browser first. Then, you can add a text editor. Then, we can do crazy stuff like time deadlines, submissions, running tests, adding new languages etc.

Some things you can look into:
- https://github.com/facebook/draft-js
- https://github.com/CompuIves/codesandbox-client

Collaboration may be little too much for now. 

An additional thing we can do is make the server run code and tests, kinda like how prairielearn does it. 
Just keep in mind our server runs in serverless using [zeit now](https://zeit.co/now) so storing files with code and running them would possibly not work. Python, Javascript, and other scripting languages would also be a lot easier to run than compiled ones. (A language-agnostic compiler in python https://github.com/HaliteChallenge/Halite/blob/master/worker/compiler.py)

I'd make this another folder in the main directory (same level as the folders `frontend` and `backend`)

**Comment by rgychiu** [comment]
There's another component that we could use, but I'll have to look into this one more. What are your thoughts on https://github.com/securingsincity/react-ace

**Comment by rgychiu** [comment]
I also feel like in terms of customizability, React Ace has a lot of potential (just from a basic glance through), but obviously the easiest route would just be to slap in the codesandbox since it has basically everything already done.

**Comment by tko22** [comment]
yep I like that! I added codesandbox just as a good reference since it's pretty good and very smooth.
